python/eo_acquisition.py

```diff
@@ -416,6 +423,195 @@ def compute_exptime(self, target_counts, meas_flux):
+        where shifted clear drops/raises P-High before/after clearing
+        and where the shifted value "phi_shifted" is hardcoded below
+        """
+        if (nclears < 1 ):
+            return
+        ccdtype = self.get_ccdtype()
+        if ccdtype == 'e2v':
+            phi_shifted = 5.5  #- hard coded value for now
+            #- verify input value makes sense
+            #
+            if (phi_shifted < 5.0 or phi_shifted > 7.0):
+                self.logger.info("P-HI:{} not in range 5.0..7.0".format(phi_shifted))
+                    #- throw an exception here or something
+                raise java.lang.Exception("Bad phi_shifted value {}".format(phi_shifted))
+            #
+            #- get original values for Parallel high and low rails
+            #
+            phi = self.getParallelHighConfig()
+            if phi == None:
+                self.logger.info("getParallelHighConfig() = None")
+                #- throw an exception here or something
+                raise java.lang.Exception("failed getting PHi config")
+            #
+            plo = self.getParallelLowConfig()
+            if plo == None:
+                self.logger.info("getParallelLowConfig() = None")
+                #- throw an exception here or something
+                raise java.lang.Exception("failed getting PLow config")
+            #
+            #- determine operating mode (unipolar (3) or bipolar (0))
+            #
+	    self.logger.info("phi[]= {}".format(phi))
+	    self.logger.info("plo[]= {}".format(plo))
+            cnt = 0
+            for i in range(len(phi)):
+                if phi[i] > 7.5 and plo[i] >= 0.0  and plo[i] < 2.0:
+                    cnt += 1
+                if phi[i] > 2.0 and phi[i] < 7.0  and plo[i] < -3.0:
+                    cnt -= 1
+            if cnt == 3:
+                unipolar = True
+                self.logger.info("Parallel Voltages are unipolar => Shifted Clearing")
+            elif cnt == -3:
+                unipolar = False
+                self.logger.info("Parallel Voltages are bipolar => Regular Clearing")
+            else:
+                raise java.lang.Exception(
+                    "invalid mode: must be bipolar (+/-) or unipolar (>0)")
+            if unipolar:
+                #
+                #- change to the new value
+                #
+                self.logger.info("changing dac {} to {}...".format(
+                    "pclkHighP", phi_shifted))
+                self.sub.ts8dac0.synchCommand(10, "change", "pclkHighP", phi_shifted)
+                self.sub.ts8dac1.synchCommand(10, "change", "pclkHighP", phi_shifted)
+                self.sub.ts8dac2.synchCommand(10, "change", "pclkHighP", phi_shifted)
+                self.sub.ts8.synchCommand(10, "loadDacs true")
+                #
+        #- Perform the Clear main
+        #
+        self.logger.info("Clearing CCD {} times...".format(nclears))
+        for seqno in range(nclears):
```

Comment:
Since `seqno` isn't used in this `if` block, the recommended practice is to use `_` as the dummy variable for the loop.

Reply:
yes, thanks